### PR TITLE
Feature/se 1129.add tests for zero coupon bond

### PR DESCRIPTION
**Pull request by lorenzdec**
# Pull Request Checklist

- [ ] Read the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] Tests pass
- [ ] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: sdk/tests/tutorials/instruments/test_complex_instruments.py [hunk 1]
```diff
@@ -106,3 +106,70 @@ def test_create_fx_option(self):
...
+                reset_calendars=[],
+                settle_days=1,
+                reset_days=0
+                ),
+            rate=0.03,
+            instrument_type="TermDeposit"
+        )
+        upsert_term_deposit = self.instruments_api.upsert_instruments(request_body={
+        "upsert_request_001": models.InstrumentDefinition(
+            name="Term Deposit Test3%",
```
Comment (jz-finbourne) [inline sdk/tests/tutorials/instruments/test_complex_instruments.py:133]: Me being nosy/pedantic (sorry): I think we should name it "Term Deposit Example". That's how I understand the Team SE approach anyway.

Similar on line 134.

As I presume this code to showcase (than actually test).

File: sdk/tests/tutorials/instruments/test_complex_instruments.py [hunk 2]
```diff
@@ -106,3 +106,70 @@ def test_create_fx_option(self):
...
+                reset_days=2
+                ),
+            identifiers={},
+            instrument_type="Bond"
+        )
+
+        upsert_zero_coupon_bond = self.instruments_api.upsert_instruments(request_body={
+            "upsert_request_001": models.InstrumentDefinition(
+                name="Zero Coupon Test",
+                identifiers={"ClientInternal": models.InstrumentIdValue("test_zc")},
```
Comment (jz-finbourne) [inline sdk/tests/tutorials/instruments/test_complex_instruments.py:169]: Think "ZCB" is the more common abbreviation?

**Comment by rizwansaeed** [comment]
remember to raise against develop not master

File: sdk/tests/tutorials/instruments/test_complex_instruments.py [hunk 3]
```diff
@@ -106,3 +106,70 @@ def test_create_fx_option(self):
...
+                payment_calendars=[],
+                reset_calendars=[],
+                settle_days=1,
+                reset_days=0
+                ),
+            rate=0.03,
+            instrument_type="TermDeposit"
+        )
+        upsert_term_deposit = self.instruments_api.upsert_instruments(request_body={
+        "upsert_request_001": models.InstrumentDefinition(
```
Comment (inwaves) [inline sdk/tests/tutorials/instruments/test_complex_instruments.py:132]: To avoid using the string literal "upsert_request_001" in both the API call and in the test assertion at the end of the method, this could be stored in a variable, `request_id = "upsert_request_001"` and replaced in all occurrences.

File: sdk/tests/tutorials/instruments/test_complex_instruments.py (hunk 1 above)
Comment (inwaves) [reply sdk/tests/tutorials/instruments/test_complex_instruments.py:133]: +1, I think we generally try to avoid "demo" and "test" in our examples across the sample notebooks and SDK tests.

File: sdk/tests/tutorials/instruments/test_complex_instruments.py [hunk 4]
```diff
@@ -106,3 +106,70 @@ def test_create_fx_option(self):
...
+            name="Term Deposit Test3%",
+            identifiers={"ClientInternal": models.InstrumentIdValue("test_td")},
```
Comment (inwaves) [inline sdk/tests/tutorials/instruments/test_complex_instruments.py:134]: `test_td` -> `TermDepositInstrumentCode`

File: sdk/tests/tutorials/instruments/test_complex_instruments.py (hunk 2 above)
Comment (inwaves) [reply sdk/tests/tutorials/instruments/test_complex_instruments.py:169]: As above for the "test" identifiers: `test_zc` -> `ZeroCouponBondInstrument` and `name = "Zero Coupon Bond Example"

**Review by inwaves — COMMENTED** [review]
Looks much better, good job @lorenzdec.

I'm thinking that it would be useful to tidy up once the tests are done by deleting the two instruments you've created. You could do something like:

```
# Define the instrument name at the beginning of the test as part of set-up
term_deposit_identifier = "TermDepositInstrument" 

# Later, once test has passed
self.instruments_api.delete_instrument("ClientInternal", term_deposit_identifier)
```

Ping me if you get stuck and I can give you a hand.
